### PR TITLE
Cyber network support

### PR DESCRIPTION
**Pull request by darcang**
adding support for Cyber L2 network https://docs.cyber.co/build-on-cyber/network-info.

To add some context, we got into a situation when our staked tokens could not be unstacked because Tangem is not supporting their new network, to which they pushed everyone(me included). Happy to have any feedback to move this forward!

### CODE REVIEW

File: BlockchainSdk/Blockchains/Ethereum/OtherChains/CyberExternalLinkProvider.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import Foundation
+
+
+struct CyberExternalLinkProvider: ExternalLinkProvider {
+    let testnetFaucetURL = URL(string: "https://faucet.quicknode.com/cyber/sepolia")
+    private let baseExplorerUrl: String
+
+    init(isTestnet: Bool) {
+        baseExplorerUrl = isTestnet ? "https://cyber-testnet.alt.technology" : "https://cyber.alt.technology"
```
Comment (dbaturin) [inline BlockchainSdk/Blockchains/Ethereum/OtherChains/CyberExternalLinkProvider.swift:17]:
```suggestion
        baseExplorerUrl = isTestnet ? "https://testnet.cyberscan.co/" : "https://cyberscan.co/"
```

**Review by dbaturin — CHANGES_REQUESTED** [review]

File: BlockchainSdk/Blockchains/Ethereum/OtherChains/CyberExternalLinkProvider.swift (hunk 1 above)
Comment (darcang) [reply BlockchainSdk/Blockchains/Ethereum/OtherChains/CyberExternalLinkProvider.swift]: https://github.com/tangem/blockchain-sdk-swift/pull/756/commits/d7202f579e9230893dc10fb552a7ddf3683c92e3

File: BlockchainSdk.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "originHash" : "09ecb99db3e580b585d5d932184af81c65d50399b08821d46e7de1ac03c0fdd2",
+  "originHash" : "a0738665bda10cbdb05da546422ce445e2fdf9ace530c7941d4cba18d71a3d25",
```
Comment (tureck1y) [inline BlockchainSdk.xcworkspace/xcshareddata/swiftpm/Package.resolved:2]: This file should be reverted

File: BlockchainSdk/Common/Blockchain.swift [hunk 3]
```diff
@@ -79,6 +79,7 @@ public indirect enum Blockchain: Equatable, Hashable {
     case joystream(curve: EllipticCurve)
     case bittensor(curve: EllipticCurve)
     case koinos(testnet: Bool)
+    case cyber(testnet: Bool)
```
Comment (tureck1y) [inline BlockchainSdk/Common/Blockchain.swift:82]: supportsEIP1559 is missing. Basically we check it by eth_feeHistory response. You can attach the response here and we will check

```
{
  "params": [
     5
    "latest",
    [25, 50, 75]
  ],
  "id": 1,
  "jsonrpc": "2.0",
  "method": "eth_feeHistory"
}
```

**Review by tureck1y — COMMENTED** [review]
Thanks for your amazing work. We will accept this PR and take this network for internal testing after we prepare the icons and our backend.

**Comment by tureck1y** [comment]
Cyber is an L2 network, so we must check how the L1 fee is calculated. 

Probably, the EthereumOptimisticRollupWalletAssembly should be used. 

https://docs.cyber.co/cyber-l2-and-optimism-superchain

File: BlockchainSdk.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 2 above)
Comment (darcang) [reply BlockchainSdk.xcworkspace/xcshareddata/swiftpm/Package.resolved]: https://github.com/tangem/blockchain-sdk-swift/pull/756/commits/6d8875ac933c941c93721c9dbb3de0a119d815e4

File: BlockchainSdk/Common/Blockchain.swift (hunk 3 above)
Comment (darcang) [reply BlockchainSdk/Common/Blockchain.swift:82]:
```json
{
   "jsonrpc":"2.0",
   "id":1,
   "result":{
      "oldestBlock":"0x3707f7",
      "reward":[
         [
            "0x0",
            "0xf4240",
            "0xf4240"
         ],
         [
            "0x0",
            "0x0",
            "0x0"
         ],
         [
            "0xf4240",
            "0xf4240",
            "0xf4240"
         ],
         [
            "0x0",
            "0x0",
            "0xf4240"
         ],
         [
            "0xf4240",
            "0xf452c",
            "0xf452c"
         ]
      ],
      "baseFeePerGas":[
         "0xfc",
         "0xfc",
         "0xfc",
         "0xfc",
         "0xfc",
         "0xfc"
      ],
      "gasUsedRatio":[
         0.0035375333333333334,
         0.0014605,
         0.0073422,
         0.0024133,
         0.012337633333333334
      ]
   }

**Comment by darcang** [comment]
@tureck1y thanks for the feedback!
Yeah that's what i [assumed](https://github.com/tangem/blockchain-sdk-swift/pull/756/files#diff-e0f9d65d5cc30a869c673655ea6fff26a5194a702430c16bdfdc5dd28814c5d0R1199) as well in analogy to base network

**Comment by darcang** [comment]
@tureck1y Also if there is anything I could do to speed this up a bit, like opening PRs for adding icons and backend support, please let me know

File: BlockchainSdk/Common/API/TestnetAPINodeInfoProvider.swift [hunk 4]
```diff
@@ -205,6 +205,9 @@ struct TestnetAPINodeInfoProvider {
             return [
                 .init(url: URL(string: "https://harbinger-api.koinos.io")!)
             ]
+        case .cyber:
+            return [
+                .init(url: URL(string: "https://cyber-testnet.alt.technology")!)
```
Comment (amuraveinik) [inline BlockchainSdk/Common/API/TestnetAPINodeInfoProvider.swift:210]: There is `]` missing

File: BlockchainSdk/Common/Blockchain.swift (hunk 3 above)
Comment (amuraveinik) [reply BlockchainSdk/Common/Blockchain.swift:82]: According to the response, the network should support EIP-1559.

Please fetch the latest changes to your PR, set the corresponding property value to true, and after that, we'll be able to run tests.

**Review by AlekseiMuraveinik1 — CHANGES_REQUESTED** [review]

File: BlockchainSdk/Common/API/TestnetAPINodeInfoProvider.swift (hunk 4 above)
Comment (darcang) [reply BlockchainSdk/Common/API/TestnetAPINodeInfoProvider.swift:210]: https://github.com/tangem/blockchain-sdk-swift/pull/756/commits/539d4038ba0fd33818a04125d84fd305ac27fa6d

File: BlockchainSdk/Blockchains/Ethereum/OtherChains/CyberExternalLinkProvider.swift [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+import Foundation
+
+
+struct CyberExternalLinkProvider: ExternalLinkProvider {
+    let testnetFaucetURL = URL(string: "https://faucet.quicknode.com/cyber/sepolia")
+    private let baseExplorerUrl: String
+
+    init(isTestnet: Bool) {
+        baseExplorerUrl = isTestnet ? "https://testnet.cyberscan.co/" : "https://cyberscan.co/"
```
[the remark this answers is not in the document]
Comment (darcang) [reply BlockchainSdk/Blockchains/Ethereum/OtherChains/CyberExternalLinkProvider.swift]: https://github.com/tangem/blockchain-sdk-swift/pull/756/commits/fe06f1fb3e8f56d3b3f67cf38f2aec668bd45629

File: BlockchainSdk/Common/Blockchain.swift (hunk 3 above)
Comment (darcang) [reply BlockchainSdk/Common/Blockchain.swift:83]: [this](https://github.com/tangem/blockchain-sdk-swift/blob/fe06f1fb3e8f56d3b3f67cf38f2aec668bd45629/BlockchainSdk/Common/Blockchain.swift#L751) one already set to true 
Is there another place i should change?

**Review by AlekseiMuraveinik1 — CHANGES_REQUESTED** [review]

**Comment by AlekseiMuraveinik1** [comment]
Additionally, please add case `cyber` to the `currencyId` computed property, just as it is done for `base`. 
This is to ensure the price is displayed correctly in the app

**Comment by AlekseiMuraveinik1** [comment]
> Additionally, please add case `cyber` to the `currencyId` computed property, just as it is done for `base`. This is to ensure the price is displayed correctly in the app

```swift
 /// Should be used to get the actual currency rate.
    public var currencyId: String {
        switch self {
        case .arbitrum(let testnet),
             .optimism(let testnet),
             .aurora(let testnet),
             .manta(let testnet),
             .zkSync(let testnet),
             .polygonZkEVM(let testnet),
             .base(let testnet):
             // <- case .cyber
            return Blockchain.ethereum(testnet: testnet).coinId
        default:
            return coinId
        }
    }
```

File: BlockchainSdk/Common/Blockchain.swift [hunk 6]
```diff
@@ -1160,6 +1168,8 @@ private extension Blockchain {
             return "koinos"
         case .internetComputer:
             return "internet-computer"
+        case .cyber:
+            return "cyber"
```
[the remark this answers is not in the document]
Comment (darcang) [reply BlockchainSdk/Common/Blockchain.swift]: https://github.com/tangem/blockchain-sdk-swift/pull/756/commits/986f14324284ae5c4fe07e123487ec639d60f1b0

File: BlockchainSdk/Common/Blockchain.swift [hunk 7]
```diff
@@ -748,6 +754,7 @@ extension Blockchain {
         case .flare: return true
         case .taraxa: return false
         case .base: return true
+        case .cyber: return true
```
Comment (amuraveinik) [inline BlockchainSdk/Common/Blockchain.swift:757]:
```suggestion
        case .cyber: return false
```

Comment (amuraveinik) [reply BlockchainSdk/Common/Blockchain.swift:757]: I apologize for the error in one of my previous comments about eip1559 support. It seems that setting this property to true enables sending transactions, but not with the maximum amount. Using the maximum amount results in an error related to gas * price, indicating that eip1559 is not supported

**Review by AlekseiMuraveinik1 — CHANGES_REQUESTED** [review]

**Review by AlekseiMuraveinik1 — APPROVED** [review]

**Review by dbaturin — APPROVED** [review]

**Review by fedorov-d — APPROVED** [review]

**Review by tureck1y — APPROVED** [review]

**Review by m3g0byt3 — APPROVED** [review]

**Pull request merged**